source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -1805,11 +1808,24 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
             processInst(globalInst);
         }
 
+        if (targetCaps.implies(CapabilityAtom::SPV_KHR_vulkan_memory_model))
+        {
+            m_sharedContext->m_memoryModel = SpvMemoryModelVulkan;
+        }
+
         if (m_sharedContext->m_spvVersion < 0x10300)
         {
             // Direct SPIRV backend does not support generating SPIRV before 1.3,
             // we will issue an error message here.
             m_sharedContext->m_sink->diagnose(SourceLoc(), Diagnostics::spirvVersionNotSupported);
+
```

Comment:
I wonder how one would actually trigger this case, because if coopmatrix capability requires spirv 1.6, then we can;'t end up with spirv version == 1.3 here, due to auto profile upgrade, right?